crates/storage/src/transactional.rs

```diff
@@ -190,13 +229,24 @@ where
     /// Commits the changes into the storage.
     pub fn commit(mut self) -> StorageResult<Storage> {
         let changes = core::mem::take(&mut self.inner.changes);
-        self.inner.storage.commit_changes(changes)?;
+        self.inner
+            .storage
+            .commit_changes(changes, self.inner.height_policy)?;
         Ok(self.inner.storage)
     }
 }
 
 impl<Storage> Modifiable for InMemoryTransaction<Storage> {
-    fn commit_changes(&mut self, changes: Changes) -> StorageResult<()> {
+    /// Panics height policy mismatches the height policy fo the transaction.
+    fn commit_changes(
+        &mut self,
+        changes: Changes,
+        height_policy: ModifyHeightPolicy,
```

Comment:
Internal details of how the database works should not affect the interfaces.

Reply:
I rewrote the PR to use a flag in the Database itself. The solution seems quite clean. Let me know what you think about this one.